woocommerce/woocommerce-android
# Conditional request plugins data

‼️ This PR depends on [this FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2509)

Closes: #7242 

### Description
When the order details screen is opened, the app always fetches Shipping Labels and Shipment Tracking information even if the store doesn't have those plugins installed/activated.  The idea of this PR is to request this information only when we know it is needed. To achieve this, we are now looking for the plugin's information on the local database and refreshing the data only when they are present and activated. 
There could be an edge case of trying to request for the plugin's information at a moment when the database is not filled yet. To understand better how common this scenario is, we added a track event, so we can evaluate if we need to iterate on this solution.

### Testing instructions
TC1
1. Select a site with Shipping Labels and Shipment Tracking plugins
2. Open the orders list
3. Open an order details
4. Check on Flipper that the app fetches Shipping Labels and Shipment Tracking information:

`/wc/v2/orders/671/shipment-trackings/&_method=get`
`/wc/v1/connect/label/671/&_method=get`
`/wc/v1/connect/label/671/creation_eligibility/&_method=get`

TC2
1. Select a site without Shipping Labels and Shipment Tracking plugins
2. Open the orders list
3. Open an order details
4. Check on Flipper that the app DOESN'T fetch Shipping Labels and Shipment Tracking information

### Images/gif
#### Request without Shipping Labels and Shipment Tracking plugins

<img width="1016" alt="Screen Shot 2022-08-19 at 11 19 54" src="https://user-images.githubusercontent.com/18119390/185639326-d8cc72eb-afc6-4af2-9c99-47e5873eccd7.png">

#### Request with Shipping Labels and Shipment Tracking plugins
<img width="1015" alt="Screen Shot 2022-08-19 at 11 20 39" src="https://user-images.githubusercontent.com/18119390/185639334-d8c29266-d908-4268-84d2-8e4f1fcc8b28.png">
